operate/bridge/providers/relay_bridge_provider.py

```diff
@@ -0,0 +1,445 @@
+                timestamp=int(time.time()),
+            )
+            bridge_request.quote_data = quote_data
+            bridge_request.status = BridgeRequestStatus.QUOTE_DONE
+            return
+
+        url = "https://api.relay.link/quote"
+        headers = {"Content-Type": "application/json"}
+        payload = {
+            "originChainId": Chain(from_chain).id,
+            "user": from_address,
+            "originCurrency": from_token,
+            "destinationChainId": Chain(to_chain).id,
+            "recipient": to_address,
+            "destinationCurrency": to_token,
+            "amount": to_amount,
+            "tradeType": "EXACT_OUTPUT",
+            "enableTrueExactOutput": False,
+        }
+        for attempt in range(1, DEFAULT_MAX_QUOTE_RETRIES + 1):
+            start = time.time()
+            try:
+                self.logger.info(f"[RELAY PROVIDER] POST {url}")
+                self.logger.info(
+                    f"[RELAY PROVIDER] BODY {json.dumps(payload, indent=2, sort_keys=True)}"
+                )
+                response = requests.post(
+                    url=url, headers=headers, json=payload, timeout=30
+                )
+                response.raise_for_status()
+                response_json = response.json()
+
+                # Gas will be returned as 0 (unable to estimate) by the API endpoint when simulation fails.
+                # This happens when 'from_address'
+                #   * does not have enough funds/ERC20,
+                #   * requires to approve an ERC20 before another transaction.
+                # Call the API again using the default 'from_address' placeholder used by Relay DApp.
+                gas_missing = any(
+                    "gas" not in item.get("data", {})
+                    for step in response_json.get("steps", [])
+                    for item in step.get("items", [])
+                )
+
+                if gas_missing:
+                    placeholder_payload = payload.copy()
+                    placeholder_payload["user"] = GAS_ESTIMATE_FALLBACK_ADDRESS
+                    self.logger.info(f"[RELAY PROVIDER] POST {url}")
+                    self.logger.info(
+                        f"[RELAY PROVIDER] BODY {json.dumps(placeholder_payload, indent=2, sort_keys=True)}"
+                    )
+                    placeholder_response = requests.post(
+                        url=url, headers=headers, json=placeholder_payload, timeout=30
+                    )
+                    response_json_placeholder = placeholder_response.json()
+
+                    for i, step in enumerate(response_json.get("steps", [])):
+                        for j, item in enumerate(step.get("items", [])):
+                            if "gas" not in item.get("data", {}):
+                                placeholder_gas = (
+                                    response_json_placeholder.get("steps", [])[i]
+                                    .get("items", [])[j]
```

Comment:
if "steps" or "items" doesn't exist, it will fail with a KeyError, right?
```suggestion
                                    response_json_placeholder.get("steps", {i: {}})[i]
                                    .get("items", {j: {}})[j]
```

Reply:
yeah, but I mean if the `response_json_placeholder` doesn't have the `"steps"` key, then it will return an empty array `[]` (default of `.get()`). And it will try to get the `[i]` from empty array, raising an IndexError.